src/Kiota.Builder/Plugins/PluginsGenerationService.cs

```diff
@@ -104,6 +106,131 @@ public async Task GenerateManifestAsync(CancellationToken cancellationToken = de
         }
     }
 
+    private static OpenApiDocument InlineRequestBodyAllOf(OpenApiDocument openApiDocument)
+    {
+        if (openApiDocument.Paths is null) return openApiDocument;
+        foreach (var path in openApiDocument.Paths)
+        {
+            if (path.Value?.Operations is null) continue;
+            foreach (var operation in path.Value.Operations)
+            {
+                if (operation.Value?.RequestBody?.Content is null) continue;
+                foreach (var contentItem in operation.Value.RequestBody.Content)
+                {
+                    var schema = contentItem.Value.Schema;
+                    // Merge all schemas in allOf `schema.MergeAllOfSchemaEntries()` doesn't seem to do the right thing.
+                    schema = MergeAllOfInSchema(schema);
+                    schema = SelectFirstAnyOfOrOneOf(schema);
+                    contentItem.Value.Schema = schema;
+                }
+            }
+        }
+
+        return openApiDocument;
+
+        static OpenApiSchema? SelectFirstAnyOfOrOneOf(OpenApiSchema? schema)
+        {
+            if (schema?.AnyOf is not { Count: > 0 } && schema?.OneOf is not { Count: > 0 }) return schema;
+            OpenApiSchema newSchema;
+            if (schema.AnyOf is { Count: > 0 })
+            {
+                newSchema = schema.AnyOf[0];
+            }
+            else if (schema.OneOf is { Count: > 0 })
+            {
+                newSchema = schema.OneOf[0];
+            }
+            else
+            {
+                newSchema = schema;
+            }
+            return newSchema;
+        }
+        static OpenApiSchema? MergeAllOfInSchema(OpenApiSchema? schema)
+        {
+            if (schema?.AllOf is not { Count: > 0 }) return schema;
+            var newSchema = new OpenApiSchema();
+            foreach (var apiSchema in schema.AllOf)
+            {
+                if (apiSchema.Title is not null) newSchema.Title = apiSchema.Title;
+                if (!string.IsNullOrEmpty(apiSchema.Type))
+                {
+                    if (!string.IsNullOrEmpty(newSchema.Type) && newSchema.Type != apiSchema.Type)
+                    {
+                        throw new InvalidOperationException(
+                            $"The schemas in allOf cannot have different types: '{newSchema.Type}' and '{apiSchema.Type}'.");
+                    }
+                    newSchema.Type = apiSchema.Type;
+                }
+                if (apiSchema.Format is not null) newSchema.Format = apiSchema.Format;
```

Comment:
I'm not sure how. We're changing the shape of the schema.

Reply:
I don't see a way to do this that reduces code. I'm trying to merge multiple schemas here and a copy constructor only supports creating a new schema from an existing one. I'd still need the manual assignment for all other schemas after the first one, no?

Reply:
I guess it might save you time for scalar properties of the schema, but yes, you'd still need to update/adjust the other properties (any/one/all of, etc...)

Reply:
How would you use a copy constructor here?

Reply:
ah nevermind, I had misread the code thinking newSchema was in scope of the foreach. But it's outside.